File: examples/complex/process-bar/python/_cott_impl/foo/bar/process_bar.py
from cott_runtime import Err, Ok, Result
from foo.bar import build_output, process_payload_bytes, validate_payload
from foo.bar_types import BarError, BarOptions, InputPayload, OutputPayload


def process_bar(data: InputPayload, options: BarOptions) -> Result[OutputPayload, BarError]:
    validated = validate_payload(data)
    if isinstance(validated, Err):
        return validated
    processed = process_payload_bytes(validated.value.data, options)
    if isinstance(processed, Err):
        return processed
    return Ok(value=build_output(processed.value, data.declared_size, data.format))

File: examples/kotlin/complex/process-bar/kotlin/cott_impl/foo/bar/process_bar.kt
package cott_impl.foo.bar

internal fun process_bar(`data`: foo.bar.InputPayload, options: foo.bar.BarOptions): cott_runtime.CottResult<foo.bar.OutputPayload, foo.bar.BarError> {
    val validated = when (val result = foo.bar.validate_payload(data)) {
        is cott_runtime.Ok -> result.value
        is cott_runtime.Err -> return result
    }
    val processed = when (val result = foo.bar.process_payload_bytes(validated.data, options)) {
        is cott_runtime.Ok -> result.value
        is cott_runtime.Err -> return result
    }
    return cott_runtime.Ok(foo.bar.build_output(processed, data.declared_size, data.format))
}
